packages/language/src/language-server/hover-request.ts

```diff
@@ -355,28 +359,40 @@ const generateNameTokenMarkup: MarkupGenerator = ({ unit, token }) => {
  * Generates hover content based on the provided generators and context.
  * Returns an array of all non-null results from all generators.
  */
-function generateMarkup(
+async function generateMarkup(
   generators: MarkupGenerator[],
   context: MarkupGeneratorContext,
-): MarkupResponse[] {
-  const tryGenerate = (generator: MarkupGenerator) => {
-    try {
-      return generator(context);
-    } catch {
-      return null;
-    }
-  };
-
-  return generators
-    .map(tryGenerate)
-    .filter((response): response is string => response !== null);
+): Promise<MarkupResponse[]> {
+  const results = await Promise.all(
+    // REVIEW THIS LATER
+    generators.map(async (gen) => {
+      try {
+        return await gen(context);
```

Comment:
No need to await if we want these to run concurrently, but I'm curious if there was a reason for wanting to rewrite this? If there wasn't an issue with it before I would favor leaving it in its original form.